codespell: add config, workflow + have a few typos fixed by it

yarikoptic opened the pull request.

yarikoptic committed:
Add rudimentary codespell config

yarikoptic committed:
Add github action to codespell main on push and PRs

jfnielsen commented:
Apologies, I just renamed the branches and in the process realized I hadn't dealt with this pull request. Since the branch structure has changed since your original PR would it be safer to issue a new PR?

yarikoptic committed:
[DATALAD RUNCMD] run codespell throughout fixing typos automagically (but ignoring overall fail due to ambigous ones)

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "codespell -w || :",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^

yarikoptic committed:
[DATALAD RUNCMD] chore: run codespell throughout fixing a few typos interactively

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "codespell -w -i 3 -C 4",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^
